src/graph/heterograph.cc

```diff
@@ -260,6 +261,145 @@ HeteroGraphPtr HeteroGraph::CopyTo(HeteroGraphPtr g, const DLContext& ctx) {
                                         hgindex->num_verts_per_type_));
 }
 
+HeteroGraphPtr HeteroGraph::CopyToSharedMem(
+      HeteroGraphPtr g, const std::string& name, const std::set<std::string>& fmts) {
+  auto hg = std::dynamic_pointer_cast<HeteroGraph>(g);
+  if (hg->shared_mem_name_ == name)
+    return g;
+  
+  std::string buf;
+  dmlc::MemoryStringStream ofs(&buf);
```

Comment:
You can use MemoryFixedSizeStream if the size is known (avoid one copy here)

Reply:
> You can use MemoryFixedSizeStream if the size is known (avoid one copy here)

The size of serialized meta graph is unknown

Reply:
But I see the file size you created is SHARED_MEM_METAINFO_SIZE_MAX, which means you can create first and directly write into it with MemoryFixedSizeStream